netbox/netbox/api/viewsets/__init__.py

```diff
@@ -157,3 +159,22 @@ def perform_destroy(self, instance):
         logger.info(f"Deleting {model._meta.verbose_name} {instance} (PK: {instance.pk})")
 
         return super().perform_destroy(instance)
+
+
+class MPTTLockedMixin(GenericViewSet):
+    """
+    Puts pglock on objects that derive from MPTTModel for parallel API calling.
+    Note: If adding this to a view, must add the model name to ADVISORY_LOCK_KEYS
+    """
+
+    def create(self, request, *args, **kwargs):
+        with advisory_lock(ADVISORY_LOCK_KEYS[self.queryset.model._meta.verbose_name.lower().replace(" ", "")]):
```

Comment:
The `model_name` meta attribute works for this:
```suggestion
        with advisory_lock(ADVISORY_LOCK_KEYS[self.queryset.model._meta.model_name]):
```